### PR TITLE
FB-435 new transactions for orders

### PR DESCRIPTION
**Pull request by zolt**

### CODE REVIEW

File: transactions/storefront/sell.cdc [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import RaribleNFT from "../../../contracts/RaribleNFT.cdc" //template
```
Comment (zhalninde) [inline transactions/storefront/sell.cdc:1]: импорты надо поправить

File: transactions/storefront/update_price.cdc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import FlowToken from "../../../contracts/core/FlowToken.cdc" //template
+import FungibleToken from "../../../contracts/core/FungibleToken.cdc"
+import NonFungibleToken from "../../../contracts/core/NonFungibleToken.cdc"
+import NFTStorefront from "../../../contracts/core/NFTStorefront.cdc"
+
+transaction(
+    orderId: UInt64,
+    tokenId: UInt64,
+    price: UFix64, 
+    originFees: {Addres: UFix64}, //additional fees from other marketplace etc.
```
Comment (zhalninde) [inline transactions/storefront/update_price.cdc:11]: %s/Addres/Address/g

File: transactions/storefront/update_price.cdc [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import RaribleNFT from "../../../contracts/RaribleNFT.cdc" //template
```
Comment (zhalninde) [inline transactions/storefront/update_price.cdc:1]: тоже импорты, не проходит компиляцию

**Review by zhalninde — CHANGES_REQUESTED** [review]

File: transactions/storefront/buy.cdc [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import FlowToken from "../../../contracts/core/FlowToken.cdc" //template (vault Type)
+import FungibleToken from "../../../contracts/core/FungibleToken.cdc"
+import NFTStorefront from "../../../contracts/core/NFTStorefront.cdc"
+import NonFungibleToken from "../../../contracts/core/NonFungibleToken.cdc"
+
+/*
+ * Buy item
+ * 
+ */
+transaction(orderId: UInt64, storefrontAddress: Address, fees: {Addres: UFix64}) {
```
[the remark this answers is not in the document]
Comment (zolt) [reply transactions/storefront/buy.cdc]: пофиксил, вроде везде

File: transactions/storefront/buy.cdc [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+/*
+ * Buy item
+ * 
+ */
+transaction(orderId: UInt64, storefrontAddress: Address, fees: {Address: UFix64}) {
+    let paymentVault: @FungibleToken.Vault
+    let feeVault: @FungibleToken.Vault
+    let storefront: &NFTStorefront.Storefront{NFTStorefront.StorefrontPublic}
+    let tokenReceiver: &{NonFungibleToken.Receiver}
+    let feeReceivers: {Capability<&{FungibleToken.Receiver}>: UFix64}
```
Comment (vitalbadjo) [inline transactions/storefront/buy.cdc:16]: this line throw error: cannot use type as dictionary key type: `Capability<&AnyResource{FungibleToken.Receiver}>`
18 |     let feeReceivers: {Capability<&{FungibleToken.Receiver}>: UFix64}
     |.                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

File: transactions/storefront/buy.cdc [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+import FungibleToken from "../../contracts/core/FungibleToken.cdc"
+import NFTStorefront from "../../contracts/core/NFTStorefront.cdc"
+import NonFungibleToken from "../../contracts/core/NonFungibleToken.cdc"
+
+/*
+ * Buy item
+ * 
+ */
+transaction(orderId: UInt64, storefrontAddress: Address, fees: {Address: UFix64}) {
+    let paymentVault: @FungibleToken.Vault
```
Comment (vitalbadjo) [inline transactions/storefront/buy.cdc:12]: error: cannot use incompletely initialized value: `self`
  --> d4251a0c407d491f9c38910efbb31cdaa152a2c2983e0951f6913e0 ...  `paymentVault` of type `Transaction` is not invalidated (moved or destroyed)
  --> d4251a0c407d491f9c38910efbb31cdaa152a2c2983e0951f6913e0beb607111:14:8
   |
14 |     let paymentVault: @FungibleToken.Vault
   |         ^^^^^^^^^^^^ not invalidated

File: transactions/storefront/buy.cdc [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+    prepare(acct: AuthAccount) {
+        self.storefront = getAccount(storefrontAddress)
+            .getCapability(NFTStorefront.StorefrontPublicPath)!
+            .borrow<&NFTStorefront.Storefront{NFTStorefront.StorefrontPublic}>()
+            ?? panic("Could not borrow Storefront from provided address")
+
+        let listing = self.storefront.borrowListing(listingResourceID: orderId)
+                    ?? panic("No Offer with that ID in Storefront")
+        let price = self.listing.getDetails().salePrice
```
Comment (vitalbadjo) [inline transactions/storefront/buy.cdc:26]: has no member listing, "self" is redundant

**Review by atapin — APPROVED** [review]

**Review by zhalninde — APPROVED** [review]

**Pull request merged**